distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/KadAction.java

```diff
@@ -0,0 +1,379 @@
+package com.gruppo1.distributednetworkmanager;
+
+import androidx.annotation.NonNull;
+
+import com.dezen.riccardo.networkmanager.StringResource;
+import com.dezen.riccardo.smshandler.SMSMessage;
+import com.dezen.riccardo.smshandler.SMSPeer;
+import com.dezen.riccardo.smshandler.exceptions.InvalidMessageException;
+
+import java.util.BitSet;
+
+/***
+ * Class defining an Action travelling through a Kademlia Network.
+ * Syntax for a Message is as follows:
+ * [ACTION TYPE] [OPERATION ID] [PART K] [MAX PARTS] [PAYLOAD TYPE] [PAYLOAD]
+ * @author  Pardeep Kumar, Riccardo De Zen
+ */
+
+public class KadAction implements DistributedNetworkAction<String, SMSPeer, SMSMessage> {
+    private static final int OPERATION_ID_LENGTH=4;
+    private static final int CURRENT_PART_LENGTH=4;
+    private static final int TOTAL_PARTS_LENGTH=4;
+    private static final int ACTION_TYPE_START_INDEX=0;
+    private static final int OPERATION_ID_START_INDEX=1;
+    private static final int CURRENT_PART_START_INDEX=5;
+    private static final int TOTAL_PARTS_START_INDEX=9;
+    private static final int PAYLOAD_TYPE_START_INDEX=13;
+    private static final int PAYLOAD_START_INDEX=14;
+    private static final char PARSING_CHARACTER='0';
+    // The length of the Node_ID
+    private static final int ID_LENGTH = 128;
+    public static final int MIN_ID = 1;
+    public static final int MAX_ID = 999;
+    private static final String RESOURCE_SEPARATOR="\r";
```

Comment:
Also this, it should be public, or the asshole outside (being me) can't build resource payload
Anyway we should also add a better parsing strategy